src/ucp/wireup/select.c

```diff
@@ -603,15 +596,18 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
             UCS_STATIC_BITMAP_AND_INPLACE(&rsc_addr_index_map, addr_index_map);
         }
 
-        wireup_info[0] = '\0';
+        if (info_str != NULL) {
+            info_str[0] = '\0';
+        }
```

Comment:
can we remove?

Reply:
I wanted to reset it somewhere to prevent printing trash at the end. It could happen if all the transports fail to wireup in early stages before any is_reachable function is called (for example if `uct_iface_is_reachable_params_addrs_valid` fails and we don't print anything to info_str). I think I can remove this part and maybe just initialize the string when declaring it (here actually https://github.com/openucx/ucx/pull/9995#discussion_r1683132250)

Reply:
we can initialize it to empty string in uct_iface_is_reachable_v2() and then transports could (possibly) override